js/main.js

```diff
@@ -21,7 +21,7 @@ function enterZip(event) {
 function submit(event) {
   if (
     event.key === 'Enter' &&
-    Number($zipCodeInput.value)
+    Number($zipCodeInput.value && weatherData.zipCode !== null)
   ) {
     $loading.className = 'fas fa-moon loading';
     setTimeout(spinIcon, 1000);
```

Comment:
You shouldn't do this with a `setTimeout`, you should call this function in the `load` event handler for the request. This is not an exact measure of how long you need to wait, you really want to wait just as long as the request takes.